src/view/LoginView.java

```diff
@@ -0,0 +1,145 @@
+import interface_adapter.signup.SignupController;
+import interface_adapter.signup.SignupViewModel;
+import interface_adapter.ViewManagerModel;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+
+public class LoginView extends JPanel implements ActionListener, PropertyChangeListener {
+
+    public final String viewName = "log in";
+    private final LoginViewModel loginViewModel;
+    private final SignupViewModel signupViewModel;
+    private final ViewManagerModel viewManagerModel;
+
+
+    final JTextField usernameInputField = new JTextField(15);
+    private final JLabel usernameErrorField = new JLabel();
+
+    final JPasswordField passwordInputField = new JPasswordField(15);
+    private final JLabel passwordErrorField = new JLabel();
+
+    final JButton signUp;
+    final JButton logIn;
+//    final JButton cancel;
+    private final LoginController loginController;
+    private final SignupController signupController;
+
+    public LoginView(LoginViewModel loginViewModel, LoginController controller, SignupViewModel signupViewModel, SignupController signupController, ViewManagerModel viewManagerModel) {
+
+        this.loginController = controller;
+        this.loginViewModel = loginViewModel;
+        this.loginViewModel.addPropertyChangeListener(this);
+        this.signupController = signupController;
+        this.signupViewModel = signupViewModel;
+        this.viewManagerModel = viewManagerModel;
+
+
+        JLabel title = new JLabel("Login Screen");
+        title.setAlignmentX(Component.CENTER_ALIGNMENT);
+
+        LabelTextPanel usernameInfo = new LabelTextPanel(
+                new JLabel("Username"), usernameInputField);
+        LabelTextPanel passwordInfo = new LabelTextPanel(
+                new JLabel("Password"), passwordInputField);
+
+        JPanel buttons = new JPanel();
+        logIn = new JButton(loginViewModel.LOGIN_BUTTON_LABEL);
+        buttons.add(logIn);
+        signUp = new JButton(loginViewModel.GO_TO_SIGNUP_BUTTON_LABEL);
+        buttons.add(signUp);
+
+        logIn.addActionListener(                // This creates an anonymous subclass of ActionListener and instantiates it.
+                this
+        );
+        signUp.addActionListener(
+                // This creates an anonymous subclass of ActionListener and instantiates it.
+                this
+        );
+
+//        cancel.addActionListener(this);
+
+        usernameInputField.addKeyListener(new KeyListener() {
+            @Override
+            public void keyTyped(KeyEvent e) {
+                LoginState currentState = loginViewModel.getState();
+                currentState.setUsername(usernameInputField.getText() + e.getKeyChar());
+                loginViewModel.setState(currentState);
+            }
```

Comment:
There are these keyPressed methods that don't seem to do anything here, as well as in passwordInputField.